base/src/main/java/org/aya/tyck/StmtTycker.java

```diff
@@ -137,6 +138,8 @@ public void tyckHeader(@NotNull Decl decl, @NotNull ExprTycker tycker) {
         // It might contain unsolved holes, but that's acceptable.
         var resultRes = tycker.synthesize(fn.result).wellTyped().freezeHoles(tycker.state);
         fn.signature = new Def.Signature(resultTele, resultRes);
+        if (resultTele.isEmpty() && fn.body.isRight() && fn.body.getRightValue().isEmpty())
+          reporter.report(new NobodyError(decl.sourcePos, fn.ref));
```

Comment:
Can we do this check before tycking the telescope? So `def a a` would be rejected nicely.

Reply:
`def a a` does not have empty telescope.

Reply:
If your telescope uses yourself, you should reject the function

Reply:
Yes. My brain must be offline. 😢